src/utils.rs

```diff
@@ -16,22 +21,37 @@ pub fn to_binary(value: usize, n_bits: usize) -> Vec<bool> {
     result
 }
 
+// TODO(Gotti): n_bits is a misnomer if base > 2. Should be n_limbs or sth.
+// Also, should the behaviour for value >= base^n_bits be specified as part of the API or asserted not to happen?
+// Currently, we compute the decomposition of value % (base^n_bits).
+
+/// decomposes value into its big-endian base-ary decomposition, meaning we return a vector v, s.t.
+///
+/// value = v[0]*base^(n_bits-1) + v[1] * base^(n_bits-2) + ... + v[n_bits-1] * 1,
+/// where each v[i] is in 0..base.
+/// The returned vector always has length exactly n_bits (we pad with leading zeros);
 pub fn base_decomposition(value: usize, base: u8, n_bits: usize) -> Vec<u8> {
     // Initialize the result vector with zeros of the specified length
     let mut result = vec![0u8; n_bits];
 
     // Create a mutable copy of the value for computation
+    // Note: We could just make the local passed-by-value argument `value` mutable, but this is clearer.
     let mut value = value;
 
     // Compute the base decomposition
     for i in 0..n_bits {
         result[n_bits - 1 - i] = (value % (base as usize)) as u8;
         value /= base as usize;
     }
+    // TODO: Should we assert!(value == 0) here to check that the orginally passed `value` is < base^n_bits ?
 
     result
 }
 
+// Gotti: Consider renaming this function. The name sounds like it's a PRG.
+// TODO (Gotti): Check that ordering is actually correct at point of use (everything else is big-endian).
```

Comment:
This randomness can actually be in whatever order we want, as long as it is self-consistent